jetty-util/src/main/java/org/eclipse/jetty/util/MultiException.java

```diff
@@ -31,11 +31,28 @@
 public class MultiException extends Exception
 {
     private List<Throwable> nested;
+    boolean superConstructorCalled = false;
 
     /* ------------------------------------------------------------ */
     public MultiException()
     {
         super("Multiple exceptions");
+        this.superConstructorCalled = true;
+    }
+    
+    /**
+     * We only want to fill in the stack trace if we have an exception to throw.
+     * 
+     * This prevents the stack trace from being written when the constructor is
+     * called, yet will let as fill it in later on e.g. when we actually want
+     * to throw the multi exception.
+     */
+    @Override
+    public Throwable fillInStackTrace() {
```

Comment:
I don't really like calling a virtual method from a super constructor.... I know it can work, but it's kind of wrong.

Reply:
It is a hack because the outer class is of the wrong type. The outer class should really be a `MultiExceptionBuilder` not an exception and it should throw a `MultiException`.